src/client/conn/http1.rs

```diff
@@ -439,6 +441,24 @@ impl Builder {
         self
     }
 
+    /// Set the maximum number of headers.
+    ///
+    /// When a response is received, the parser will reserve a buffer to store headers for optimal
+    /// performance.
+    ///
+    /// If client receives more headers than the buffer size, the error "message header too large"
+    /// is returned.
+    ///
+    /// Note that headers is allocated on the stack by default, which has higher performance. After
```

Comment:
Is there a need to include this paragraph? While it is interesting, perhaps we don't need really need it...

(Same with server).

Reply:
I feel that as a user, this paragraph can clearly tell me the behavior of this API and help me make decisions.

Otherwise, I might just set a very large value (whether it is used or not).

Reply:
If this option has any impact on performance, I think it needs to be explained.

Reply:
That's fair! I suppose my concern was about it "promising" how internals worked. If we later updated this to be able to allocate less. But thinking this through as I respond, I don't think anyone would mind if  a later update made things faster. It's not like they would be depending on it allocating on the heap or stack, just that it's fast or why it's less fast.

Seems good!